k8s-fluentd/main.tf

```diff
@@ -0,0 +1,32 @@
+locals {
+  # this should be the default calculated name anyway, but we want to enforce it just to be sure
+  service_account_name = var.release == "fluentd" ? "fluentd" : "${var.release}-fluentd"
+
+  # extract data about kubernetes cluster
+  cloud            = var.kubernetes_cluster.specs.kubernetes.cloud
+  k8s_distribution = var.kubernetes_cluster.specs.kubernetes.distribution
+
+  helm_values = {
+    serviceAccount = {
+    name = local.service_account_name
+    }
+    labels = var.md_metadata.default_tags
+    plugins = ["fluent-plugin-opensearch"]
+  }
+}
+
+resource "helm_release" "fluentd" {
+  name             = var.release
+  chart            = "fluentd"
```

Comment:
FYI, looking at those cloudwatch docs made me realize that fluent is now offering fluentbit as a lightweight alternative.

As far as I understand, its a bit like logstash vs filebeat (fluentd vs fluentbit) whereas the former is bulkier (logstash is java, fluentd is ruby) with more processing and configuration options, and the latter is just a lightweight log shipper. It sounds like the configuration is pretty consistent between fluentd and fluentbit. If we can use fluentbit and get the lightweight version that's probably best.  It might just be as simple as changing the chart here to `fluentbit` and the version below to something appropriate for that chart.